src/main/java/io/cdap/plugin/gcp/bigtable/sink/RecordToHBaseMutationTransformer.java

```diff
@@ -0,0 +1,88 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.gcp.bigtable.sink;
+
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.format.UnexpectedFormatException;
+import io.cdap.cdap.api.data.schema.Schema;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.util.Bytes;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Transforms Google Cloud Bigtable {@link Result} to {@link StructuredRecord}.
+ */
+public class RecordToHBaseMutationTransformer {
+
+  private final String keyAlias;
+  private final String columnFamily;
+
+  public RecordToHBaseMutationTransformer(String keyAlias, String columnFamily) {
+    this.keyAlias = keyAlias;
+    this.columnFamily = columnFamily;
+  }
+
+  public Mutation transform(StructuredRecord record) {
+    List<Schema.Field> fields = Objects.requireNonNull(record.getSchema().getFields(), "Schema fields cannot be empty");
+    String key = Objects.requireNonNull(record.get(keyAlias), "Key cannot be null");
+    Put put = new Put(Bytes.toBytes(key));
+    byte[] columnFamilyBytes = Bytes.toBytes(columnFamily);
+    for (Schema.Field field : fields) {
+      String fieldName = field.getName();
+      try {
+        byte[] valueBytes = convertFieldValueToBytes(record.get(fieldName), field.getSchema());
+        put.addColumn(columnFamilyBytes, Bytes.toBytes(fieldName), valueBytes);
+      } catch (Exception e) {
+        throw new UnexpectedFormatException(String.format("Failed to transform value for field '%s'. Reason: %s",
+                                                          fieldName, e.getMessage()));
+      }
+    }
+    return put;
+  }
+
+  private byte[] convertFieldValueToBytes(Object value, Schema fieldSchema) {
+    if (fieldSchema.isNullable()) {
+      if (null == value) {
+        return null;
+      }
+      return convertFieldValueToBytes(value, fieldSchema.getNonNullable());
+    }
+    switch (fieldSchema.getType()) {
+      case NULL:
+        return null;
+      case STRING:
+        return Bytes.toBytes((String) value);
+      case BYTES:
+        return (byte[]) value;
+      case INT:
+        return Bytes.toBytes((int) value);
+      case LONG:
+        return Bytes.toBytes((long) value);
+      case FLOAT:
+        return Bytes.toBytes((float) value);
+      case DOUBLE:
+        return Bytes.toBytes((double) value);
+      case BOOLEAN:
+        return Bytes.toBytes((boolean) value);
+      default:
+        throw new UnexpectedFormatException("Field type '" + fieldSchema.getType() + "' is not supported");
```

Comment:
should pass in a Schema.Field into the method so that we can include the field name in this error message:

"Field '%s' is of unsupported type '%s'".

Reply:
Field name is provided in the caller method. I wrapped convertFieldValueToBytes to handle possible ClassCastExceptions.